tensorflow/core/kernels/data/cache_dataset_ops_test.cc

```diff
@@ -43,446 +102,238 @@ class CacheDatasetOpTest : public DatasetOpsTestBase {
   }
 
  protected:
-  // Creates `TensorSliceDataset` variant tensor from the input vector of
-  // tensors.
-  Status CreateTensorSliceDatasetTensor(
-      std::vector<Tensor>* const tensor_vector, Tensor* dataset_tensor) {
-    DatasetBase* tensor_slice_dataset;
-    TF_RETURN_IF_ERROR(CreateTensorSliceDataset(
-        "tensor_slice_node", tensor_vector, &tensor_slice_dataset));
-    TF_RETURN_IF_ERROR(
-        StoreDatasetInVariantTensor(tensor_slice_dataset, dataset_tensor));
-    return Status::OK();
-  }
-
-  // Create a new `CacheDataset` op kernel.
-  Status CreateCacheDatasetOpKernel(
-      const DataTypeVector& output_types,
-      const std::vector<PartialTensorShape>& output_shapes,
-      std::unique_ptr<OpKernel>* cache_dataset_op_kernel) {
-    NodeDef node_def = test::function::NDef(
-        kNodeName, name_utils::OpName(CacheDatasetOp::kDatasetType),
-        {CacheDatasetOp::kInputDataset, CacheDatasetOp::kFileName},
-        {{CacheDatasetOp::kOutputTypes, output_types},
-         {CacheDatasetOp::kOutputShapes, output_shapes}});
-    TF_RETURN_IF_ERROR(CreateOpKernel(node_def, cache_dataset_op_kernel));
-    return Status::OK();
-  }
-
-  // Create a new `CacheDataset` op kernel context.
-  Status CreateCacheDatasetContext(
-      OpKernel* const op_kernel,
-      gtl::InlinedVector<TensorValue, 4>* const inputs,
-      std::unique_ptr<OpKernelContext>* context) {
-    TF_RETURN_IF_ERROR(CheckOpKernelInput(*op_kernel, *inputs));
-    TF_RETURN_IF_ERROR(CreateOpKernelContext(op_kernel, inputs, context));
-    TF_RETURN_IF_ERROR(ParseScalarArgument<tstring>(
-        context->get(), CacheDatasetOp::kFileName, &filename_));
-    return Status::OK();
-  }
-
- private:
-  tstring filename_ = "";
-};
-
-struct TestCase {
-  std::vector<Tensor> input_tensors;
-  string file_name;
-  std::vector<Tensor> expected_outputs;
-  DataTypeVector expected_output_dtypes;
-  std::vector<PartialTensorShape> expected_output_shapes;
-  int64 expected_cardinality;
-  std::vector<int> breakpoints;
+  tstring cache_filename_;
```

Comment:
`cache_filename_ ` is used to store the input cache file name, so we can delete the cache files in the destructor after each test task finishes. 

Although `dataset_ctx_` stores the pointer to the input tensor of `filename`, the `filename` tensor is held by the input dataset params, which has been deallocated before the destructor runs. That means, the input tensors for the op can not be accessed in the destructor.